src/main/java/managers/ZAPManager.java

```diff
@@ -0,0 +1,91 @@
+package managers;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.zaproxy.clientapi.core.ApiResponse;
+import org.zaproxy.clientapi.core.ApiResponseElement;
+import org.zaproxy.clientapi.core.ClientApi;
+import org.zaproxy.clientapi.core.ClientApiException;
+import utilities.PropertiesLoader;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class ZAPManager {
+    private static final Logger log = LogManager.getLogger(ZAPManager.class.getName());
+    private PropertiesLoader propertiesLoader = new PropertiesLoader();
+    private ClientApi clientApi;
+
+    public ZAPManager() {
+        String zapAddress = propertiesLoader.getZAPAddress();
+        String zapPort = propertiesLoader.getZAPPort();
+        String zapApiKey = propertiesLoader.getZAPApiKey();
+        if (zapAddress == null | zapPort == null | zapApiKey == null) {
+            return;
+        }
+        this.clientApi = new ClientApi(zapAddress,
+                Integer.parseInt(zapPort), zapApiKey);
+        log.info("Proxy client Api created");
+    }
+
+    public ClientApi getClientApi() {
+        return clientApi;
+    }
+
+    public void spiderScan(String targetUrl) {
+        try {
+            //set it to prevent freezing scanner
+            clientApi.spider.setOptionMaxDepth(1);
```

Comment:
@kontownik 's comment: maxDepth could be parameter set in feature + info about it in README

Reply:
added